crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,223 @@
+    pub context: FileSystemPathVc,
+}
+
+#[turbo_tasks::value_impl]
+impl Issue for UnsupportedSwcEcmaTransformPluginsIssue {
+    #[turbo_tasks::function]
+    fn severity(&self) -> IssueSeverityVc {
+        IssueSeverity::Warning.into()
+    }
+
+    #[turbo_tasks::function]
+    fn category(&self) -> StringVc {
+        StringVc::cell("transform".to_string())
+    }
+
+    #[turbo_tasks::function]
+    async fn title(&self) -> Result<StringVc> {
+        Ok(StringVc::cell(format!(
+            "Unsupported SWC Ecma transform plugins on this platform."
+        )))
+    }
+
+    #[turbo_tasks::function]
+    fn context(&self) -> FileSystemPathVc {
+        self.context
+    }
+
+    #[turbo_tasks::function]
+    fn description(&self) -> StringVc {
+        StringVc::cell(
+            "Turbopack does not yet support running SWC ecma transform plugins on this platform."
+                .to_string(),
+        )
+    }
+}
+
+#[derive(Debug)]
+pub struct SwcEcmaTransformPluginsTransformer {
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    plugins: Vec<(SwcPluginModuleVc, serde_json::Value)>,
+}
+
+impl SwcEcmaTransformPluginsTransformer {
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    pub fn new(plugins: Vec<(SwcPluginModuleVc, serde_json::Value)>) -> Self {
+        Self { plugins }
+    }
+
+    #[cfg(not(feature = "swc_ecma_transform_plugin"))]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl CustomTransformer for SwcEcmaTransformPluginsTransformer {
+    #[cfg_attr(not(feature = "swc_ecma_transform_plugin"), allow(unused))]
+    async fn transform(&self, program: &mut Program, ctx: &TransformContext<'_>) -> Result<()> {
+        #[cfg(feature = "swc_ecma_transform_plugin")]
+        {
+            use std::{cell::RefCell, rc::Rc, sync::Arc};
+
+            use swc_core::{
+                common::{
+                    comments::SingleThreadedComments,
+                    plugin::{
+                        metadata::TransformPluginMetadataContext, serialized::PluginSerializedBytes,
+                    },
+                    util::take::Take,
+                },
+                ecma::ast::Module,
+                plugin::proxies::{HostCommentsStorage, COMMENTS},
+                plugin_runner::plugin_module_bytes::PluginModuleBytes,
+            };
+
+            let mut plugins = vec![];
+            for (plugin_module, config) in &self.plugins {
+                let plugin_module = &plugin_module.await?.0;
+
+                plugins.push((
+                    plugin_module.get_module_name().to_string(),
+                    config.clone(),
+                    Box::new(plugin_module.clone()),
+                ));
+            }
+
+            let should_enable_comments_proxy =
+                !ctx.comments.leading.is_empty() && !ctx.comments.trailing.is_empty();
+
+            //[TODO]: as same as swc/core does, we should set should_enable_comments_proxy
```

Comment:
Do we have an issue tracking this? If so, add the ticket number here.

Reply:
There isn't. Also it need some audit if we're going to make transformcontext to pass Option<Comment> to allow this or not.